framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/KieServerWithExternalDatabaseScenarioImpl.java

```diff
@@ -58,37 +54,37 @@ public KieServerWithExternalDatabaseScenarioImpl(OpenShiftController openShiftCo
     }
 
     @Override public void deploy() {
-        project = createProject(openshiftController);
+        project = createProject();
 
         logger.info("Processing template and creating resources from " + OpenShiftTemplate.KIE_SERVER_DATABASE_EXTERNAL.getTemplateUrl().toString());
         envVariables.put(OpenShiftTemplateConstants.IMAGE_STREAM_NAMESPACE, project.getName());
         project.processTemplateAndCreateResources(OpenShiftTemplate.KIE_SERVER_DATABASE_EXTERNAL.getTemplateUrl(), envVariables);
 
-        kieServerDeployment = new KieServerDeploymentImpl();
-        kieServerDeployment.setOpenShiftController(openshiftController);
-        kieServerDeployment.setNamespace(project.getName());
+        kieServerDeployment = new KieServerDeploymentImpl(project);
         kieServerDeployment.setUsername(DeploymentConstants.getKieServerUser());
         kieServerDeployment.setPassword(DeploymentConstants.getKieServerPassword());
 
         logger.info("Waiting for Kie server deployment to become ready.");
         kieServerDeployment.waitForScale();
     }
 
-    @Override public void undeploy() {
-        InstanceLogUtil.writeDeploymentLogs(this);
+    @Override
+    public void undeploy() {
```

Comment:
This same cleanup logic seems to be repeated in several scenarios. Would it make sense to extract it to some utility class?

Reply:
I extracted undeploy method and part of deploy method to parent abstract OpenShiftScenario.